generate-test-vector.py

```diff
@@ -1419,6 +1387,125 @@ def generate_uncompressed_keys_tests():
+
+    outpoints = [
+        ("f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16", 0),
+        ("f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16", 1),
+        ("f4184fc596403b9d638783cf57adfe4c75c605f6356fbc91338530e9831e9e16", 2)
+    ]
+    sender_bip32_seed = 'deadbeef'
+    recipient_bip32_seed = 'f00dbabe'
+    scan, spend, Scan, Spend = reference.derive_silent_payment_key_pair(bytes.fromhex(recipient_bip32_seed))
+    address = reference.encode_silent_payment_address(Scan, Spend, hrp=HRP)
+    addresses = [(address, 1.0)]
+
+    sender['given']['recipients'] = addresses
+    recipient['given']['key_material']['scan_priv_key'] = scan.get_bytes().hex()
+    recipient['given']['key_material']['spend_priv_key'] = spend.get_bytes().hex()
+    recipient['expected']['addresses'] = [address]
+
+    inputs = []
+    input_priv_keys = []
+    input_pub_keys = []
+        
+    # p2sh(p2wpkh) compressed
+    i = len(inputs)
+    priv, pub = get_key_pair(i, seed=bytes.fromhex(sender_bip32_seed))
+    sig = priv.sign_ecdsa(msg, low_s=False, rfc6979=True).hex()
+    p2wpkh = "0014"+reference.hash160(pub.get_bytes(False)).hex()
+    p2sh = f"a914{hash160(bytes.fromhex(p2wpkh)).hex()}87"
+    inputs += [{
+        'txid': outpoints[i][0],
+        'vout': outpoints[i][1],
+        'scriptSig': f'{(len(p2wpkh)//2):0x}'+p2wpkh,
+        'txinwitness': serialize_witness_stack([sig, pub.get_bytes(False).hex()]),
+        'prevout': { 'scriptPubKey': { 'hex': p2sh } }
+    }]
+    input_priv_keys += [(priv, False)]
+    input_pub_keys += [pub]
+
+    # p2sh(p2wpkh) uncompressed
+    i = len(inputs)
+    priv, pub = get_key_pair(i, seed=bytes.fromhex(sender_bip32_seed))
+    pub.compressed = False
+    sig = priv.sign_ecdsa(msg, low_s=False, rfc6979=True).hex()
+    p2wpkh = "0014"+reference.hash160(pub.get_bytes(False)).hex()
+    p2sh = f"a914{hash160(bytes.fromhex(p2wpkh)).hex()}87"
+    inputs += [{
+        'txid': outpoints[i][0],
+        'vout': outpoints[i][1],
+        'scriptSig': f'{(len(p2wpkh)//2):0x}'+p2wpkh,
+        'txinwitness': serialize_witness_stack([sig, pub.get_bytes(False).hex()]),
+        'prevout': { 'scriptPubKey': { 'hex': p2sh } }
+    }]
+    input_priv_keys += [(priv, False)]
+
+    # p2sh(p2ms)
+    keys = [
+        get_key_pair(0, seed=bytes.fromhex(sender_bip32_seed)),
+        get_key_pair(1, seed=bytes.fromhex(sender_bip32_seed))
+    ]
+    # OP_2 pub1 pub2 OP_3 OP_EQUAL
+    multisig_script = "52" + ''.join(["21"+key[1].get_bytes(False).hex() for key in keys]) + "53ae"
+    p2sh = f"a914{hash160(bytes.fromhex(multisig_script)).hex()}87"
+
+    sigs = [key[0].sign_ecdsa(msg, low_s=False, rfc6979=True).hex() for key in keys]
+    sigs = [ f'{(len(sig)//2):0x}' + sig + "01" for sig in sigs]
+
+    i = len(inputs)
+    inputs += [{
+        'txid': outpoints[i][0],
+        'vout': outpoints[i][1],
+        'scriptSig': f"00{''.join(sigs)}4c{(len(multisig_script)//2):0x}{multisig_script}",
```

Comment:
why do you have OP_PUSHDATA1 here? Is it necessary?

Reply:
Actually, it's not. I just checked my P2MS transactions and they are not there. It seems I included the opcode erroneously

Reply:
> ~Actually, it's not. I just checked my P2MS transactions and they are not there. It seems I included the opcode erroneously~

@S3RK Sorry, I got confused momentarily. I'm not sure if it's necessary but my P2MS transaction's `scriptSig` has a `OP_PUSHDATA1` for the redeem script.

Reply:
ah, makes sense. We need it since we're probably pushing more than 75 bytes. Resolving.